src/vs/workbench/services/files/node/fileService.ts

```diff
@@ -315,6 +315,29 @@ export class FileService implements IFileService {
 		});
 	}
 
+	public setContentsAndResolve(resource: uri, absolutePath: string, value: string, addBom: boolean, encodingToWrite: string, options: { encoding?: string; mode?: number; flag?: string; }): TPromise<IFileStat> {
+		let writeFilePromise: TPromise<void>;
+
+		// Write fast if we do UTF 8 without BOM
+		if (!addBom && encodingToWrite === encoding.UTF8) {
+			options.encoding = encoding.UTF8;
+			writeFilePromise = pfs.writeFile(absolutePath, value, options);
+		}
+
+		// Otherwise use encoding lib
+		else {
+			const encoded = encoding.encode(value, encodingToWrite, { addBOM: addBom });
```

Comment:
@seishun oh yeah I see, then maybe just rename the parameter in this method so that you can use the shorthand.